### PR TITLE
fixes infinitely repeating LDPA search results with PHP <= 7.2

### LINKED ISSUE #20745: Repeated user results on search page (LDAP, PHP 7.2)
(This is mostly a reminder for me)

### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Have LDAP configured and PHP up to 7.2 (old API)
2. Go to users page
3. Do a search

### Expected behaviour

* Looked for user is being shown once

### Actual behaviour

* Looked for user is being shown infinitely

### PR DESCRIPTION
**Pull request by blizzz**
fixes #20745

### CODE REVIEW

**Review by MorrisJobke — APPROVED** [review]

**Review by rullzer — APPROVED** [review]
:elephant:

**Review by juliusknorr — APPROVED** [review]
Code looks good, however I fail to reproduce the issue on my PHP 7.2 instance, so I cannot verify that it actually works :see_no_evil:

File: apps/user_ldap/lib/Access.php [hunk 1]
```diff
@@ -2049,6 +2049,12 @@ private function initPagedSearch(
 				// of '0' is valid, because 389ds
 				$reOffset = ($offset - $limit) < 0 ? 0 : $offset - $limit;
 				$this->search($filter, $base, $attr, $limit, $reOffset, true);
+				if (!$this->hasMoreResults()) {
+					// when the cookie is reset with != 0 offset, there are no further
+					// results, so stop. This if block is not necessary with new API
+					// and can be removed with dropping PHP 7.2
```
Comment (hashworks) [inline apps/user_ldap/lib/Access.php:2055]: Why can we remove this block when we drop PHP 7.2 when this solves an issue that occured on 7.4 (#20745)?
Or is that new API not implemented yet?

Comment (blizzz) [reply apps/user_ldap/lib/Access.php:2055]: I failed to reproduce it on 7.4 for some unknown reason. Ought to double check and remove this part of the comment.

**Comment by RenWal** [comment]
One word of caution for those applying this fix (and maybe something that can be automatically solved in the release):

This bug caused the `ScanFiles` job to run to an ever-increasing user offset (since it thought there are still more users because the LDAP app ignored the offset). At about offset 39000, this caused a `Maximum function nesting level of '256' reached, aborting!` since the LDAP paged user search is implemented recursively. (Monitor your log files, folks)

Now, this fix would solve the issue regarding the ever-increasing offset, but the cron job now fails before that can happen because of the recursion error. It's solved by manually forcing the offset to zero.

**Pull request merged**